### PR TITLE
Feature/main display functionality

### LINKED ISSUE #8: As a user, I should be able to Login with an Email
- [x] If I click the "Login" Button in the Header, I'm taken to the Login Page 
- [x] Login Page should have the Header
- [x] Login Page should have an input for the User's Email 
- [x] Login Page should have a Submit Button
- [x] State should update if email matches
- [x] Alert should pop up if email doesn't match

### LINKED ISSUE #9: As a user, I'm brought to a Main Page
- [x] Main page should have a header 
- [x] Main page should have a "View All Packs" button [Functional]
- [x] Main page should have a "Start New Pack" button [Disabled]
- [x] Main page should have a "Start as Lone Wolf" button [Disabled]

### LINKED ISSUE #10: As a user, I should see a Header on the Main Page
- [x] Should have Hamburger (in Mobile View Only)
- [x] Should have Name of App
- [x] Should have a fully functional "View All Liked Names" Button
- [x] Should have a fully functional Logout Button

### LINKED ISSUE #12: As a user, I shouldn't be able to type random things after the URL path
- [x] If the URL path doesn't match a specific page, I should be taken to a 404 Error Page
- [x] The 404 Error Page should have a "Back to Home" Button
- [x] Should still show Header

### LINKED ISSUE #28: As a User, I should be able to Logout
- [x] When I click Logout, state/currentUser is changed back to null
- [x]  When I click Logout, state/loggedIn is changed back to false
- [x] Add conditional for Login/Logout Button

### PR DESCRIPTION
**Pull request by EllieJellyBean**
##### Description
What's changed & where should the reviewer start?
- Added Functionality to: Login, Header buttons, Home buttons
- Added Basic styling to distinguish disabled/functional buttons
- Add routes for other built out pages ("View All Liked Names", "View All Packs", "Login")

#### Type of change
What is it?
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [x] Refactoring

### Checklist:
- [x] My code follows the guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] My changes generate no new warnings
- [x] Any dependent changes have been merged and published in downstream module

### Relevant Issues:
- Close #8 
- Close #9 
- Close #10 
- Close #12 
- Close #28

### CODE REVIEW

File: src/Components/App/App.js [hunk 1]
```diff
@@ -8,38 +7,68 @@ import Login from '../Login/Login';
...
+    this.setState({ loggedIn: true })
+  }
+
+  logoutUser = () => {
+    this.setState({ currentUser: null });
+    this.setState({ likedNames: [] });
+    this.setState({ allPacks: [] });
+    this.setState({ loggedIn: false });
+  }
+
```
Comment (zachjjohns) [inline src/Components/App/App.js:40]: nice method of resetting state, excited to see how we'll setState once we're connected to BE

File: src/Components/Header/Header.js [hunk 2]
```diff
@@ -10,9 +11,19 @@ const Header = () => {
...
         <span></span>
       </div>
+      <NavLink to='/'><button data-cy='home-button' className='home-button'>Take Me Back Home!</button></NavLink>
       <h1 className='header-title'>Pet Name Finder</h1>
-      <button>⭐️ View Liked Names ⭐️</button>
-      <button>Logout</button>
+      <NavLink to='/liked-names'><button data-cy='view-liked-button' className='view-liked-button'>⭐️ View Liked Names ⭐️</button></NavLink>
+      {!loggedIn &&
+      <NavLink to='/login'><button data-cy='login-button' className='login-button'>Login</button></NavLink>
+      }
```
Comment (zachjjohns) [inline src/Components/Header/Header.js:19]: awesome use of NavLinks here

**Review by zachjjohns — APPROVED** [review]
Awesome stuff and thank you for making quick progress!

File: src/Data/User.js [hunk 3]
```diff
@@ -3,26 +3,28 @@ const data = {
...
+            likedNames: [
                 "Daisy", "Gouda"
             ],
             packs: [2]
         },
         {
             email: "random@email.com",
             id: 3,
-            likedNamed: [
+            likedNames: [
```
Comment (Ryan-D-Miller) [inline src/Data/User.js:22]: nice catch on my mistyping

File: src/Components/Login/Login.js [hunk 4]
```diff
@@ -1,17 +1,54 @@
...
+        <input
+          className="login-input"
+          placeholder="Email"
+          value={this.state.searchInput}
+          onChange={(event) => this.handleChange(event)}
+        >
+        </input>
+        <NavLink to="/"><button
+            onClick={(event) => this.searchInput(event)}>🐾 Login Now! 🐾
+          </button></NavLink>
```
Comment (Ryan-D-Miller) [inline src/Components/Login/Login.js:48]: I like the look of this login page

**Review by Ryan-D-Miller — APPROVED** [review]

**Pull request merged**